### ISSUE TITLE
Can maximize window?

### ISSUE THREAD

**Issue by mondy** [fyne-io/fyne#776]
I want to maximize the window from the program.
And I want to know if the window is maximized.
Are these possible?
Or do you have any plans to implement it?

**Comment by andydotxyz** [comment]
Maximize has proved complicated, but it could be considered - mostly window based operations are left up to the user and/or desktop environment. i.e. your desktop could remember maximised state, so should it be overridden?

Window.SetFullScreen() is an option at the moment - is that any help?

**Comment by mondy** [comment]
Thank you for the reply.
I want to maximize window on start application. 
But I don't remember my desktop.

I think that maximized window should be remembered by each application.

> Window.SetFullScreen() is an option at the moment - is that any help?

Thanks.
But I know Window.SetFullScreen. However, it is different from the maximize window.

My environment is as follows.

* OS: Ubuntu 18.04
* go version 1.14
* fyne.io version v1.2.3

**Comment by andydotxyz** [comment]
Remembering position and maximised state is typically the job of your desktop environment (though it has more detail in #554). Modern desktops seeem to be moving to "fullscreen" for truly maximized and then various different types of "zoom" (used to be called maximize) such as "left half" etc.

So it becomes difficult to see what a user expectation is here. Can you expand on your use-case?

**Comment by mondy** [comment]
I see... I understand that it's a desktop environment job.
But what I'm looking for is not a "remember feature" but a "getter/setter feature".
For example, you may want to always maximize and start regardless of the state of the last time you quit.
For example, you might want to maximize it using a button with a widget.
"remember feature" is not necessary for fyne, but I think it should be possible to get and set the maximization state so that developers using fyne can implement it.
And it's also true that some environments don't remember position and maximised state such as my desktop environment (Ubuntu/GNOME).
I looked for a way to remember the location and maximized state in Ubuntu/GMOME, but could not find it.
I don't want to give up Ubuntu/GNOME for this.

The topic is different, but there is a possible use case that requires a getter/setter for the window position.
Think of a painting tool like GIMP, for example. Tools, layers, etc. can be displayed as a separate window. You will need to finely determine the window size and window position and align the windows. For this, we need a window position getter/setter.

In summary, I hope that the following features will be added
* window maximization setter/getter (what I need)
* window minimization setter/getter (implemented as above)
* window position setter/getter (possible use cases that need it)

**Comment by andydotxyz** [comment]
I think it would be a very, very bad idea for a window to be able to start iconified. 
For one thing you have no knowledge of what that might mean for a user - in macOS for example you would never see the window as it would be hidden behind an application icon.

Allowing applications to iconify or maximize themselves without the user having requested this action seems pretty dangerous as well - to a novice user this is what window controls are.
We have pushed this boundary with fullscreen support, but I think it's clear that fullscreen is a different modailty and applications need to be able to request that for various reasons (presentation, game screen etc).

I can see your point regarding window positioning, but if you take your example of GIMP you will see that they have moved to a single-window layout now where the pallettes are arranged around the content. This is what we too would recommend. That puts a lot more control back into the app layout. If you load an app with 3 windows into a smart phone most of the content will not be usable as you can only interact with a single window at a time.

Given our ambitions to encourage cross-platform thinking and focus on best practice I'm not sure that I'm convinced we should add the feature requests outlined above.

**Comment by mondy** [comment]
I often found that they were selecting features with a focus on best practices. Thanks for the discussion.

I'm stretching the story too far to extend the use case, but all I need is a setter/getter for window maximization. Sorry.

However, window maximization, minimization, and repositioning are standard features in other desktop GUI libraries. Also, there may come a time when someone needs it.

**Comment by andydotxyz** [comment]
Even if we wanted to do this some systems (macOS for example) require special permissions to manipulate this "outside the app" configuration. I think it would be a bad user experience if Fyne apps popped up dialogs asking for accessibility access to support these operations.

**Comment by andydotxyz** [comment]
Window management should be in the hands of the user.
We should probably do a better job of remembering their configuration as descrribed in #554.
Closing in preference of that solution.
